includes/rest-api.php

```diff
@@ -1,6 +1,54 @@
 <?php
 namespace tenup;
 
+/**
+ * Check the REST request before it's dispatched to see if it is approved for access
+ * @param WP_Error|null|bool $result The original REST result
+ * @param WP_REST_Server $rest_server Instance of the REST server
+ * @param WP_REST_Request $request The current REST request
+ * @return $result The original null request if successful or a string denying the request if it is not an approved route
+ */
+function rest_pre_dispatch($result, $rest_server, $request) {
+	$approved_routes = apply_filters( 'tenup_rest_approved_routes', $rest_server->get_routes() );
+
+	foreach( $approved_routes as $route => $data ) {
+		if( preg_match( '@^' . $route . '$@i', $request->get_route(), $matches ) ) {
+			return $result;
+		}
+	}
+
+	$result = 'your request is denied.';
+	return $result;
+}
+
+add_filter( 'rest_pre_dispatch', __NAMESPACE__ . '\rest_pre_dispatch', 3, 99);
+
+/**
+ * If desired, these endpoints can be added back by removing this filter.
+ * Alternatively, you can create a new filter to add the user endpoints back conditionally,
+ * for example with a current_user_can() permission check
+ * @param $approved_routes The list of approved routes generated by WP_REST_Server::get_routes
+ * @return $approved_routes A filtered list of the routes that removes all user endpoints by default, for security.
+ */ 
+function create_default_routes( $approved_routes ) {
+	$endpoints_to_block = array(
+		'/wp/v2/users',
+		'/wp/v2/users/(?P<id>[\d]+)',
+		'/wp/v2/users/me',
+	);
+
+	foreach ( $approved_routes as $route => $data ) {
+		if ( in_array( $route, $endpoints_to_block ) ) {
```

Comment:
We should pass `true` as the third param to `in_array` to force a strict comparison.